src/include/platform/internal/GenericPlatformManagerImpl_FreeRTOS.ipp

```diff
@@ -217,7 +217,7 @@ void GenericPlatformManagerImpl_FreeRTOS<ImplClass>::_RunEventLoop(void)
 
                 // Call into the system layer to dispatch the callback functions for all timers
                 // that have expired.
-                err = static_cast<System::LayerImplFreeRTOS &>(DeviceLayer::SystemLayer()).HandlePlatformTimer();
+                err = static_cast<System::LayerImpl &>(DeviceLayer::SystemLayer()).HandlePlatformTimer();
```

Comment:
if you fix your `LayerImplFreeRTOSSocket` inheritance to add `LayerImplFreeRTOS` you won't need this change

Reply:
As discussed, this needs to be discussed and upstreamed